### ISSUE TITLE
CTR Integration - Model Integration

### ISSUE THREAD

**Issue by ambarish-prakash** [eth-easl/modyn#88]
We need to introduce a CTR prediction model to our trainer to be able to train on the CTR data. In our case, we would like to use the DLRM prediction model from facebook. 
A tested and working DLRM model (our reference) is found here: https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/Recommendation/DLRM/README.md

We would need to integrated this model over to our trainer to be able to set it up and process the Criteo data that will be passed in from the storage, selected via the selector.

For now I see this as setting up the model code into the modyn repo and being able to instantiate it. Broadly breaking it into two independent tasks for now, but I believe more issues might come up as we start to integrate it.

TODOs:
- [ ] Review NVIDIA's implementation and remove/adapt what is not needed
- [ ] Add the model to the Modyn models directory
- [ ] Make any changes needed for data preprocessing 
- [ ] Fix Validation set?
